src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/entities/AsylumCaseBuilderTest.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.entities;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+
+import java.util.*;
+import org.junit.Test;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.AddressUk;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.IdValue;
+
+public class AsylumCaseBuilderTest {
+
+    private final Optional<String> homeOfficeReferenceNumber = Optional.of("A");
+    private final Optional<String> homeOfficeDecisionDate = Optional.of("B");
+    private final Optional<String> appellantTitle = Optional.of("C");
+    private final Optional<String> appellantGivenNames = Optional.of("D");
+    private final Optional<String> appellantLastName = Optional.of("E");
+    private final Optional<String> appellantDateOfBirth = Optional.of("F");
+    private final Optional<List<IdValue<Map<String, String>>>> appellantNationalities =
+        Optional.of(Arrays.asList(new IdValue<>("1", Collections.emptyMap())));
+    private final Optional<String> appellantHasFixedAddress = Optional.of("H");
+    private final Optional<AddressUk> appellantAddress = Optional.of(mock(AddressUk.class));
+    private final Optional<String> appealType = Optional.of("I");
+    private final Optional<String> hasNewMatters = Optional.of("J");
+    private final Optional<List<String>> newMatters = Optional.of(Arrays.asList("K"));
+    private final Optional<String> hasOtherAppeals = Optional.of("L");
+    private final Optional<List<IdValue<String>>> otherAppeals = Optional.of(Arrays.asList(new IdValue<>("1", "M")));
+    private final Optional<String> legalRepReferenceNumber = Optional.of("N");
+
+    private AsylumCaseBuilder asylumCaseBuilder = new AsylumCaseBuilder();
+
+    @Test
+    public void should_build_asylum_case() {
+
+        asylumCaseBuilder.setHomeOfficeReferenceNumber(homeOfficeReferenceNumber);
```

Comment:
Testing getters/setters can sometimes be a pain.  In the past I have used something like SmartUnit to make it less painful through reflection.  https://github.com/rlogiacco/SmartUnit/wiki/Testing-POJOs-properties.
Think there is another lib called OpenPojo or similar which I have not used.

Reply:
We are expected to use the same testing suite across all reform projects -- so that is basically JUnit and Mockito. If I had my way we would be writing beautiful Spock tests :) So I don't think we can introduce this tool.